jsonargparse_tests/test_dataclass_like.py

```diff
@@ -587,6 +718,37 @@ def test_pydantic_types(self, valid_value, invalid_value, cast, type_str, monkey
             parser.parse_args([f"--model.param={invalid_value}"])
         ctx.match("model.param")
 
+    @pytest.mark.skipif(not pydantic_supports_field_init, reason="Field.init is required")
+    def test_dataclass_field_init_false(self, parser):
+        # This tests the following error:
+        #
+        # TypeError: Parser key "data.p2":
+        # Expected a <class 'str'>. Got value: annotation=str
+        # required=False default='-' init=False
+        parser.add_argument("--data", type=PydanticDataFieldInitFalse)
+        cfg = parser.parse_args(["--data.p1=1.0"])
+        assert cfg.data.p1 == 1.0
+
+    def test_dataclass_stdlib_field(self, parser):
+        parser.add_argument("--data", type=PydanticDataStdlibField)
+        cfg = parser.parse_args(["--data.p1=1.0"])
+        assert cfg.data.p1 == 1.0
+
+    def test_dataclass_stdlib_field_init_with_factory(self, parser):
+        parser.add_argument("--data", type=PydanticDataStdlibFieldWithFactory)
+        cfg = parser.parse_args(["--data.p1=1.0"])
+        assert cfg.data.p1 == 1.0
+
+    def test_dataclass_nested(self, parser):
+        # This tests the following error:
+        #
+        # ValueError: Expected "default" argument to be an instance of
+        # "PydanticData" or its kwargs dict, given
+        # <dataclasses._MISSING_TYPE object at 0x105624c50>
```

Comment:
Same comment as above. What is this? How does this test a ValueError? If an error is tested, then I would expect an `pytest.raises` instead of some confusing comment.